sds/src/match_validation/http_validator.rs

```diff
@@ -136,71 +152,86 @@ impl HttpValidatorHelper {
 #[async_trait]
 impl MatchValidator for HttpValidator {
     async fn validate(&self, matches: &mut Vec<RuleMatch>, _: &[Box<dyn CompiledRuleDyn>]) {
-        // Let's reqwest the HTTP API endpoint to validate the matches
-        let futures = matches.iter_mut().map(|m| {
-            async move {
-                let mut request_builder: reqwest::RequestBuilder;
-                match self.config.method {
-                    HttpMethod::Get => {
-                        request_builder = HTTP_CLIENT.get(&self.config.endpoint);
-                    }
-                    HttpMethod::Post => {
-                        request_builder = HTTP_CLIENT.post(&self.config.endpoint);
-                    }
-                    HttpMethod::Put => {
-                        request_builder = HTTP_CLIENT.put(&self.config.endpoint);
-                    }
-                    HttpMethod::Delete => {
-                        request_builder = HTTP_CLIENT.delete(&self.config.endpoint);
-                    }
-                    HttpMethod::Patch => {
-                        request_builder = HTTP_CLIENT.patch(&self.config.endpoint);
+        // let's build a map of match status per endpoint and per match_idx
+        let mut match_status_per_endpoint_and_match = AHashMap::new();
+        for (idx, _) in matches.iter().enumerate() {
+            for endpoint in &self.config.endpoints {
+                match_status_per_endpoint_and_match
+                    .entry((idx, endpoint))
+                    .or_insert_with(|| MatchStatus::NotChecked);
+            }
+        }
```

Comment:
You can possibly use map/flatmap/collect methods from the iterator interface to directly create the map with the right capacity and populated items:
```suggestion
        let mut match_status_per_endpoint_and_match: AHashMap<_,_> =
       matches.iter().enumerate()
           .flat_map(|(idx, _)| {
               self.config.endpoints.iter()
                   .map(move |endpoint| ((idx, endpoint), MatchStatus::NotChecked))
        }).collect();
```